src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -85,31 +85,47 @@ public BufferedStream(Stream stream, int bufferSize)
         private void EnsureNotClosed()
         {
             if (_stream == null)
-                throw new ObjectDisposedException(null, SR.ObjectDisposed_StreamClosed);
+            {
+                Throw();
```

Comment:
Were these methods not getting inlined otherwise?

Reply:
Would you mind explaining this change some more? What's the purpose of throwing within a static internal method?

Reply:
> Were these methods not getting inlined otherwise?

They were not (I've verified that with PerfView, at least for `EnsureNotClosed`)

> Would you mind explaining this change some more? How does throwing within a static internal method solve the problem?

There are many reasons for which we won't inline a method ([old blog post](https://blogs.msdn.microsoft.com/davidnotario/2004/11/01/jit-optimizations-inlining-ii/))) but two most popular ones:

* the method is too big
* the method has exception handling (or does not return)

By moving the throw out of `EnsureNotClosed` to a local function called `Throw` (the fact that it's static does not matter here) the  `EnsureNotClosed` method gets inlined, but the `Throw` does not. So for the happy path, all the callers of `EnsureNotClosed` perform `if (_stream == null)` without a method call, but in the rare and unhappy path, they call the `Throw` method.

To see what gets inlined and what does not, you can use PerfView:

![obraz](https://user-images.githubusercontent.com/6011991/108973989-48db6d80-7685-11eb-9b69-40e2251a3e0e.png)

![obraz](https://user-images.githubusercontent.com/6011991/108974058-5abd1080-7685-11eb-9d71-aeb0fe98df51.png)

![obraz](https://user-images.githubusercontent.com/6011991/108974242-8f30cc80-7685-11eb-95e3-9e38d3aa69ae.png)